plugins/node/opentelemetry-instrumentation-net/src/instrumentation.ts

```diff
@@ -112,7 +112,12 @@ export class NetInstrumentation extends InstrumentationBase<Net> {
   ) {
     const tlsSpan = this.tracer.startSpan('tls.connect');
 
-    const netSpan = this._startSpan(options, socket);
+    const netSpan = context.with(
+      trace.setSpan(context.active(), tlsSpan),
+      () => {
```

Comment:
you could avoid this closure by passing the context directly down to `startSpan`. But maybe that's not worth the effort.

I wonder also if the net or tls span should be set as active when calling the original to ensure that e.g. a dns request done during results in a correct nested span.

Reply:
> you could avoid this closure by passing the context directly down to `startSpan`. But maybe that's not worth the effort.

I agree but that would require propagating it down to `_startSpan` and then to `_startGenericSpan`, `_startIpcSpan`, `_startTcpSpan` and I thought it would be cleaner this way

> I wonder also if the net or tls span should be set as active when calling the original to ensure that e.g. a dns request done during results in a correct nested span.

That also needs to be fixed (in many instrumentations actually, not only in net). I wrote it but had a hard time implementing unit test to assert the behavior, so decided to start with the low-hanging fruits :)

Reply:
That's fine. followup is ok.